Sources/View/NibBased.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  NibBased.swift
+//  Reusable
+//
+//  This code is under the MIT License (MIT)
+//
+//  Created by Felipe Silva on 17/04/19.
+//  Copyright © 2019 Felipe Silva . All rights reserved.
+//
+
+import UIKit
+
+/// Make your UIViewController subclasses conform to this protocol when:
+///  * they *are* NIB-based, and
+///  * this class is used as the XIB's File's Owner
+///
+/// to be able to instantiate them from the NIB in a type-safe manner
+public protocol NibBased: class {
```

Comment:
Why create a brand new protocol here while there's already one with the same semantics and purpose in the library named `NibLoadable`? I'd reuse `NibLoadable` for the purpose of this feature. The implementation you wrote below for `instantiate()` is already contained to only be available to `UIViewControllers` anyway so no risk of conflicting implementation with the usage of the protocol on Views 😉 and since the semantics and purpose is the same…

Reply:
Note: I realize I was the one who posted the code with NibBased in the first place 😅 but in retrospect I think we can avoid it 😛

Thx for the PR btw!

Reply:
I thought I was waiting for the NiBased procotol, so I did. But it's true, it can be avoided. The last commit should resolve this.

Thank you